src/sql/azdata.proposed.d.ts

```diff
@@ -644,13 +670,29 @@ declare module 'azdata' {
 
 		export type DialogWidth = 'narrow' | 'medium' | 'wide' | number;
 
+		export type DialogStyle = 'normal' | 'flyout' | 'callout' | 'calloutCompact';
+
+		export type DialogPosition = 'left' | 'below' | undefined;
+
+		export interface ITriggerProperties {
```

Comment:
This name is too generic and doesn't really seem to apply to the fields. I'd suggest calling it either something like `ICalloutDimensions` or even just `IDimensions` since it's really a pretty generic concept itself. 

But I'm also not a fan of just calling it Dimensions since that usually is just the width/height/length...don't think there's a term for both of them combined though so not sure of a better name.

Reply:
I need a property that I can pass which represents the dimensions and position of the trigger element (e.g. button). It's not part of the callout / dialog per se, but it's related in that it helps with positioning not the callout / dialog. What do you recommend?

Reply:
Keep it generic unless it's tied specifically to a concept is the general approach I try to take. So just come up with a good general name for the position + dimenions interface.